examples/angular/default-query-function/src/default-query-function-example.component.ts

```diff
@@ -0,0 +1,141 @@
+import { AngularQueryDevtoolsComponent } from '@tanstack/angular-query-devtools-experimental'
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  inject,
+  signal,
+} from '@angular/core'
+import { CommonModule, NgIf } from '@angular/common'
+import {
+  CreateQuery,
+  UseQueryClient,
+} from '@tanstack/angular-query-experimental'
+
+type Post = {
+  id: number
+  title: string
+  body: string
+}
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'post',
+  standalone: true,
+  imports: [NgIf],
+  template: `
+    <div>
+      <div>
+        <a (click)="setPostId.emit(-1)" href="#"> Back </a>
+      </div>
+      <ng-container *ngIf="postQuery().status === 'pending'">
+        Loading...
+      </ng-container>
+      <ng-container *ngIf="postQuery().status === 'error'">
+        Error: {{ postQuery().error?.message }}
+      </ng-container>
+      <ng-container *ngIf="postQuery().data as post">
+        <h1>{{ post.title }}</h1>
+        <div>
+          <p>{{ post.body }}</p>
+        </div>
+        <div *ngIf="postQuery().isFetching">Background Updating...</div>
+      </ng-container>
+    </div>
+  `,
+})
+export class PostComponent {
+  @Output() setPostId = new EventEmitter<number>()
+
+  // Until Angular supports signal-based inputs, we have to set a signal
+  @Input({ required: true })
+  set postId(value: number) {
+    this.postIdSignal.set(value)
+  }
+  postIdSignal = signal<number>(0)
+
+  // You can even leave out the queryFn and just go straight into options
+  postQuery = inject(CreateQuery)<Post>(() => ({
+    enabled: this.postIdSignal() > 0,
+    queryKey: [`/posts/${this.postIdSignal()}`],
+  }))
+
+  queryClient = inject(UseQueryClient)
```

Comment:
Why is this needed?

Reply:
The queryClient? It's used by the styling on the links to check if it's cached.

Reply:
That would be a great usecase for an angular directive to be honest 😄

Reply:
The example apps could probably be improved, or completely new ones written specific to Angular Query. On the other hand I just quickly ported them from React and it's nice if the simple ones can fit on a small amount of screen real estate inside documentation.